### PR TITLE
{Pipeline} Bump version pipeline

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

Add a new pipeline for bumping version. The related doc is also added.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
bump version pipeline

File: scripts/ci/automation_full_test.py [hunk 1]
```diff
@@ -16,11 +19,16 @@
...
+# sys.argv is passed by
+# .azure-pipelines/templates/automation_test.yml in section `Running full test`
+# scripts/bump_version/bump_version.yml in `azdev test` step
 instance_cnt = int(sys.argv[1])
 instance_idx = int(sys.argv[2])
 profile = sys.argv[3]
 serial_modules = sys.argv[4].split()
+fix_failure_tests = sys.argv[5].lower() == 'true' if len(sys.argv) >= 6 else False
+working_directory = "/home/vsts/work/1/s"
+azdev_test_result_dir = "/home/vsts/.azdev/env_config/home/vsts/work/1/s/env"
```
Comment (wangzelin007) [inline scripts/ci/automation_full_test.py:31]: Can we get these directories by pipeline variables?

File: scripts/bump_version/bump_version.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        az login -u $(CLI_LIVE_TEST_ACCOUNT) -p $(CLI_LIVE_TEST_PASSWORD)
+        az account set -s 0b1f6471-1bf0-4dda-aec3-cb9272f09590
+
+        serial_modules="appservice botservice cloud network azure-cli-core azure-cli-telemetry"
+        python scripts/ci/automation_full_test.py "8" "$(Instance_idx)" "latest" "$serial_modules" "True"
+      displayName: "Rerun tests"
+    - task: PublishTestResults@2
+      inputs:
+        testResultsFiles: '/home/vsts/.azdev/env_config/home/vsts/work/1/s/env/test_results_*.xml'
+        testRunTitle: 'CLI Bump Version test results of instance $(Instance_idx)'
```
Comment (wangzelin007) [inline scripts/bump_version/bump_version.yml:119]: Very curious, where can we see the documentation for these two parameters?

**Review by wangzelin007 — APPROVED** [review]
LGTM

File: scripts/bump_version/bump_version.yml (hunk 2 above)
Comment (evelyn-ys) [reply scripts/bump_version/bump_version.yml:119]: Here's the doc: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-test-results?view=azure-devops&tabs=trx%2Cyaml

File: scripts/ci/automation_full_test.py [hunk 3]
```diff
@@ -99,6 +107,97 @@
         }
 
 
+def run_git_command(cmd):
```
Comment (jsntcy) [inline scripts/ci/automation_full_test.py:110]: Is it better to rename the function to "run_command" as the function not only support running "git command"?

File: scripts/ci/automation_full_test.py [hunk 4]
```diff
@@ -99,6 +107,97 @@
...
+            logger.info("git push all recording files")
+            break
+        except RuntimeError as ex:
+            if retry == 0:
+                raise ex
+            retry -= 1
+            time.sleep(10)
+
+
+def run_azdev(cmd):
```
Comment (jsntcy) [inline scripts/ci/automation_full_test.py:154]: Is it better to rename the function to meaningful name as the function not only support running "azdev"?

File: scripts/ci/automation_full_test.py (hunk 3 above)
Comment (evelyn-ys) [reply scripts/ci/automation_full_test.py]: Replaced with `run_command`

File: scripts/ci/automation_full_test.py (hunk 4 above)
Comment (evelyn-ys) [reply scripts/ci/automation_full_test.py]: Replaced with `run_command`

**Review by jsntcy — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

File: scripts/bump_version/bump_version.yml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+- job: CreatePR
```
Comment (jiasli) [inline scripts/bump_version/bump_version.yml:121]: I think it'll be good to give an option to disable creating PR.

During live run, some tests may fail, so the PR created by `azclibot` cannot pass CI. The developer must manually fix those failed tests, but the developer may not have the permission to push to `azclibot` repo. In that case, the developer can add their changes onto `azclibot`'s commits and push to their own forked repo, then creating a PR from their own forked repo (for example https://github.com/Azure/azure-cli/pull/28660). The PR created by `azclibot` becomes redundant and should be closed.

**Pull request merged**